test/starknet-dss/starknetDss.ts

```diff
@@ -1,19 +1,251 @@
+	await deployer.declare(factory);
+	const pot = await deployer.deploy(factory, { vat, ward });
+	return starknetPrank<typeof potAbi>(wrapTyped(hre, pot));
+}
+
+export async function deploySNCure(
+	deployer: Account,
+	ward: Felt
+): Promise<SNCure> {
+	const factory = await hre.starknet.getContractFactory("cure");
+	await deployer.declare(factory);
+	const cure = await deployer.deploy(factory, { ward });
+	return starknetPrank<typeof cureAbi>(wrapTyped(hre, cure));
+}
+
+export async function deploySNEnd(
+	deployer: Account,
+	vat: string,
+	ward: Felt
+): Promise<SNEnd> {
+	const factory = await hre.starknet.getContractFactory("end");
+	await deployer.declare(factory);
+	const end = await deployer.deploy(factory, { ward, vat });
+	return starknetPrank<typeof endAbi>(wrapTyped(hre, end));
+}
+
+export interface SNDssInstance {
+	vat: SNVat;
+	jug: SNJug;
+	cure: SNCure;
+	dai: SNDai;
+	daiJoin: SNDaiJoin;
+	end: SNEnd;
+	spotter: SNSpotter;
+	pot: SNPot;
+}
+
+export async function deploy(
+	deployer: Account,
+	owner: Account,
+	dai: string
+): Promise<SNDssInstance> {
+	// dss.vat = VatAbstract(address(new Vat()));
+	let vat = await deploySNVat(deployer, owner.address);
+	// dss.dai = DaiAbstract(address(Dai(dai)));
+	let dai_ = await getSNDai(dai);
+	// dss.daiJoin = DaiJoinAbstract(
+	// 	address(new DaiJoin(address(dss.vat), address(dss.dai)))
+	// );
+	let daiJoin = await deploySNDaiJoin(deployer, vat.address, dai_.address);
+	// //dss.dog = DaiAbstract(address(new Dog()));  // Needs merge in xdomain-dss
+	// dss.spotter = SpotAbstract(address(new Spotter(address(dss.vat))));
+	let spotter = await deploySNSpotter(deployer, vat.address, owner.address);
+	// dss.pot = PotAbstract(address(new Pot(address(dss.vat))));
+	let pot = await deploySNPot(deployer, vat.address, owner.address);
+	// dss.jug = JugAbstract(address(new Jug(address(dss.vat))));
+	let jug = await deploySNJug(deployer, vat.address, owner.address);
+	// dss.cure = CureAbstract(address(new Cure()));
+	let cure = await deploySNCure(deployer, owner.address);
+	// dss.end = EndAbstract(address(new End(address(dss.vat))));
+	let end = await deploySNEnd(deployer, vat.address, owner.address);
+
+	// switchOwner(address(dss.vat), deployer, owner);
+	// switchOwner(address(dss.spotter), deployer, owner);
+	// switchOwner(address(dss.pot), deployer, owner);
+	// switchOwner(address(dss.jug), deployer, owner);
+	// switchOwner(address(dss.cure), deployer, owner);
+	// switchOwner(address(dss.end), deployer, owner);
+	startStarknetPrank(owner);
+
+	return {
+		vat,
+		jug,
+		dai: dai_,
+		cure,
+		daiJoin,
+		end,
+		spotter,
+		pot,
+	};
 }
 
-// TODO: add more contracts
 // TODO: add init script
 // Based on: https://github.com/makerdao/dss-bridge/blob/4cfc84761b4bfeae747af14d3a2545377dd3304a/src/deploy/XDomainDss.sol
+export async function getSNDaiJoin(address: string): Promise<SNDaiJoin> {
+	const daiJoin = await getL2ContractAt(
+		hre,
+		"dai_join.cairo/dai_join_abi.json",
```

Comment:
We should ask Shardlabs to accept abi object instead of a path.

Reply:
definitelt would make more sense